server/raft.go

```diff
@@ -1858,22 +1878,13 @@ func (n *raft) applyCommit(index uint64) error {
 		delete(n.acks, index)
 	}
 
-	// FIXME(dlc) - Can keep this in memory if this too slow.
-	ae, err := n.loadEntry(index)
-	if err != nil {
-		state := n.wal.State()
-		if index < state.FirstSeq {
-			return nil
-		}
-		if err != ErrStoreClosed && err != ErrStoreEOF {
-			if err == errBadMsg {
-				n.setWriteErrLocked(err)
-			}
-			n.warn("Got an error loading %d index: %v", index, err)
-		}
+	ae := n.pae[index]
+	if ae == nil {
```

Comment:
I assumed that you would have had a cache (not expected to maintain ALL of them), so on cache miss you would resort to call loadEntry.
Testing with this branch I see quickly the warning down here and mirrors do not complete...

Reply:
Just for the ones we have outstanding waiting to commit but I see the error loading sometimes too so will make some changes.

Reply:
Made some changes.

Reply:
Still a lot of warnings about the number of append entries pending, and on the mirror leader, "AppendEntryResponse failed to be placed on internal channel" warnings are plenty.
But at least this time the mirroring completes, but there is 1 message missing in the run I just made.
I just tried with file and same outcome (1 missing).  What I see also is that while "nats s report" shows that the lag falls to 0, it takes way more time for the count to go to the correct number, so that I believe shows the number of entries being already replicated but not yet applied?